### PR TITLE
 Add optional support for icons in check segment

### PR DESCRIPTION
**Pull request by DevelopmentCool2449**
This a improved and fixed version of this PR https://github.com/seagle0128/doom-modeline/pull/706, this should fix some bugs from original patch like icons in inactive windows, &c.

### CODE REVIEW

File: doom-modeline-segments.el [hunk 1]
```diff
@@ -846,19 +868,33 @@ mouse-2: Show help for minor mode")
...
+                                                                      (t 'doom-modeline-info))))
+                                  ('nil (format "%s/%s/%s"
+                                                (doom-modeline-check-text (number-to-string .error)
+                                                                          'doom-modeline-urgent)
+                                                (doom-modeline-check-text (number-to-string .warning)
+                                                                          'doom-modeline-warning)
+                                                (doom-modeline-check-text (number-to-string .info)
+                                                                          'doom-modeline-info)))
+                                  ('icons (format "%s %s %s %s %s"
+                                                  (doom-modeline-check-text (number-to-string .error)
```
Comment (seagle0128) [inline doom-modeline-segments.el:884]: Should handle the error icon here.

File: doom-modeline-segments.el [hunk 2]
```diff
@@ -846,19 +868,33 @@ mouse-2: Show help for minor mode")
...
-                                                                      'doom-modeline-warning)
-                                          (doom-modeline-check-text (number-to-string .info)
-                                                                      'doom-modeline-info))))))
+                                (pcase doom-modeline-check-simple-format
+                                  ('t (doom-modeline-check-text (number-to-string (+ .error .warning .info))
+                                                                (cond ((> .error 0) 'doom-modeline-urgent)
+                                                                      ((> .warning 0) 'doom-modeline-warning)
+                                                                      (t 'doom-modeline-info))))
+                                  ('nil (format "%s/%s/%s"
+                                                (doom-modeline-check-text (number-to-string .error)
```
Comment (seagle0128) [inline doom-modeline-segments.el:877]: refactor: the texts are the same below. Maybe should use `let` to simplify.

File: doom-modeline-segments.el [hunk 3]
```diff
@@ -958,8 +994,12 @@ mouse-3: Next error"
                                      ((> severity note-level)    (cl-incf .warning))
                                      (t                          (cl-incf .note))))))
                         (if (> (+ .error .warning .note) 0)
-                            (doom-modeline-check-icon "nf-md-alert_circle_outline" "❗" "!"
-                                                        (cond ((> .error 0) 'doom-modeline-urgent)
+                            (doom-modeline-check-icon (nth 0 doom-modeline-check-error-icons)
```
Comment (seagle0128) [inline doom-modeline-segments.el:997]: [L1169](https://github.com/seagle0128/doom-modeline/pull/711/files#diff-0505629a410f96e739a924cc71c2ef537a25d27a64d97615e8e7ae3d06906130R1169) `doom-modeline-display-text` may cause the incorrect displays of the icons. Did you test it?

**Review by seagle0128 — CHANGES_REQUESTED** [review]

File: doom-modeline-segments.el (hunk 3 above)
Comment (DevelopmentCool2449) [reply doom-modeline-segments.el:994]: Yes, i tested it, it works well for me, especially with flymake.
Sin título

File: doom-modeline-segments.el (hunk 1 above)
Comment (DevelopmentCool2449) [reply doom-modeline-segments.el:881]: I don't think so, found this way better, the icon error is already provided by [flymake|flycheck]-icon, which provides using their menu, however if it's required adding the error icon i would gladly do it.

File: doom-modeline-segments.el (hunk 2 above)
Comment (DevelopmentCool2449) [reply doom-modeline-segments.el]: Done

File: doom-modeline-segments.el (hunk 1 above)
Comment (seagle0128) [reply doom-modeline-segments.el]: The behaviors are different. the current error icon in `doom-modeline-update-[flymake|flycheck]-icon` may be different  color. That's why I feel the implement is not perfect.

File: doom-modeline-segments.el (hunk 3 above)
Comment (seagle0128) [reply doom-modeline-segments.el:990]: The screenshot looks in TUI. How about GUI?

File: doom-modeline-segments.el (hunk 1 above)
Comment (DevelopmentCool2449) [reply doom-modeline-segments.el]: Oh, you don't have to worrry about that, this line [933](https://github.com/seagle0128/doom-modeline/pull/711/files#diff-0505629a410f96e739a924cc71c2ef537a25d27a64d97615e8e7ae3d06906130R993) automatically don't change icon color if `doom-modeline-check-simple-format` is set to `icons` only change to 1 color `doom-modeline-urgent`.

This doesn't interfere with other check formats.

| `'nil`  | `'icons`  | `'t`  |
|---|---|---|
| 3Sin título  | 1Sin título  |  2Sin título |

File: doom-modeline-segments.el (hunk 3 above)
Comment (seagle0128) [reply doom-modeline-segments.el:990]: Of course `doom-modeline-check-icon` is a good option if it's added. But we should consider the relationship of other options.

File: doom-modeline-segments.el (hunk 1 above)
Comment (seagle0128) [reply doom-modeline-segments.el]: Did you observe the extra spaces after the icons? In the screenshots, no space after the error icon.

Comment (DevelopmentCool2449) [reply doom-modeline-segments.el]: Thank you, that was the solution i was trying to find for those extra spaces.

Now this should be fixed.
1Sin título

File: doom-modeline-segments.el (hunk 3 above)
Comment (DevelopmentCool2449) [reply doom-modeline-segments.el:990]: then Should i use `doom-modeline-icon` instead `doom-modeline-check-icon`?

If so, i should add an option for choose nerd icon kind.

Comment (seagle0128) [reply doom-modeline-segments.el:990]: We can decide this later.

**Comment by seagle0128** [comment]
I tested with the latest commit, and found the icons in the inactive windows are still not right.

image

**Comment by DevelopmentCool2449** [comment]
> I tested with the latest commit, and found the icons in the inactive windows are still not right.
> 
> image

It should now be fixed.

> And, the `doom-modeline-icon` option is not respected.
> 
> image

I don't know how to reproduce that issue, but i think it should be fixed too, it now uses `doom-modeline-display-icon` for icons and `doom-modeline-display-text` for error counters

**Comment by seagle0128** [comment]
> > I tested with the latest commit, and found the icons in the inactive windows are still not right.
> > image
> 
> It should now be fixed.
> 
> > And, the `doom-modeline-icon` option is not respected.
> > image
> 
> I don't know how to reproduce that issue, but i think it should be fixed too, it now uses `doom-modeline-display-icon` for icons and `doom-modeline-display-text` for error counters, tell me if the issue still

This issue has been fixed in the latest commit.

File: doom-modeline-segments.el (hunk 1 above)
Comment (seagle0128) [reply doom-modeline-segments.el]: Thanks. It works now. But, the help-echo and key-map are not correct yet. I am thinking it's a design issue.

**Comment by seagle0128** [comment]
I reimplement the check segment. The behavior is exactly same as VSCode. Please check and test [feat/check-icons](https://github.com/seagle0128/doom-modeline/tree/feat/check-icons) branch.
PR: https://github.com/seagle0128/doom-modeline/pull/712

`doom-modeline-check-simple-format` => nil
image

`doom-modeline-check-simple-format` => t
image

**Comment by DevelopmentCool2449** [comment]
> I reimplement the check segment. The behavior is exactly same as VSCode. Please check and test [feat/check-icons](https://github.com/seagle0128/doom-modeline/tree/feat/check-icons) branch. PR: #712
> 
> `doom-modeline-check-simple-format` => nil image
> 
> `doom-modeline-check-simple-format` => t image

Works fine, i like more that version than mine, maybe it should have "⌛" icon like original, but i don't dislike it.

**Comment by seagle0128** [comment]
Okay. I will consider the suggestion and improve it.

**Comment by seagle0128** [comment]
Closed by #712.

**Pull request closed without merging**